test/IISIntegration.FunctionalTests/Inprocess/AppOfflineTests.cs

```diff
@@ -0,0 +1,130 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Net;
+using System.Net.Http;
+using System.Threading.Tasks;
+using IISIntegration.FunctionalTests.Utilities;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Server.IISIntegration.FunctionalTests;
+using Microsoft.AspNetCore.Server.IntegrationTesting;
+using Microsoft.AspNetCore.Testing.xunit;
+using Microsoft.Extensions.Logging;
+using Xunit;
+
+namespace IISIntegration.FunctionalTests.Inprocess
+{
+    [SkipIfIISExpressSchemaMissingInProcess]
+    public class AppOfflineTests : IISFunctionalTestBase
+    {
+        [ConditionalFact]
+        public async Task AppOfflineDroppedWhileSiteIsDown_SiteReturns503()
+        {
+            var deploymentParameters = Helpers.GetBaseDeploymentParameters();
+
+            var deploymentResult = await DeployAsync(deploymentParameters);
```

Comment:
I wish there was a way to assert that site is no started yet.

Reply:
Yeah it would be a bit tricky.